Tests/ZIPFoundationTests/ZIPFoundationWritingTests+ZIP64.swift

```diff
@@ -243,60 +246,80 @@ extension ZIPFoundationTests {
         let archive = self.archive(for: #function, mode: .update)
         guard let entry = archive["data1.random"] else {
-            XCTFail("Failed to add zip64 format entry to archive"); return
+            XCTFail("Failed to retrieve zip64 format entry from archive"); return
         }
         // Should keep zip64 ecod
         do {
             try archive.remove(entry)
         } catch {
             XCTFail("Failed to remove entry from archive with error : \(error)")
         }
+        XCTAssert(archive.checkIntegrity())
         XCTAssertNotNil(archive.zip64EndOfCentralDirectory)
     }
 
     func testRemoveZIP64EntryFromArchiveWithZIP64EOCD() {
-        // File structure:
         // testRemoveEntryFromArchiveWithZIP64EOCD.zip/
         //   ├─ data1.random (size: 64)
         //   ├─ data2.random (size: 64 * 64)
         mockIntMaxValues()
         defer { resetIntMaxValues() }
         let archive = self.archive(for: #function, mode: .update)
         guard let entry = archive["data2.random"] else {
-            XCTFail("Failed to add zip64 format entry to archive"); return
+            XCTFail("Failed to retrieve zip64 format entry from archive"); return
         }
         // Should remove zip64 eocd at the same time
         do {
             try archive.remove(entry)
         } catch {
             XCTFail("Failed to remove entry from archive with error : \(error)")
         }
+        XCTAssert(archive.checkIntegrity())
         XCTAssertNil(archive.zip64EndOfCentralDirectory)
     }
 
-    // MARK: - Helpers
-
-    func resetIntMaxValues() {
-        maxUInt32 = .max
-        maxUInt16 = .max
-    }
-
-    func mockIntMaxValues(int32Factor: Int = 64, int16Factor: Int = 64) {
-        maxUInt32 = UInt32(int32Factor * int32Factor)
-        maxUInt16 = UInt16(int16Factor)
+    func testRemoveEntryWithZIP64ExtendedInformation() {
+        // testRemoveEntryFromArchiveWithZIP64EOCD.zip/
+        //   ├─ data1.random (size: 64 * 32)
+        //   ├─ data2.random (size: 64 * 32)
+        //   ├─ data3.random (size: 64 * 32) [headerID: 1, dataSize: 8, ..0..0, relativeOffsetOfLocalHeader: 4180, ..0]
+        //   ├─ data4.random (size: 64 * 32) [headerID: 1, dataSize: 8, ..0..0, relativeOffsetOfLocalHeader: 6270, ..0]
+        mockIntMaxValues()
+        defer { resetIntMaxValues() }
+        let archive = self.archive(for: #function, mode: .update)
+        guard let entry2 = archive["data2.random"] else {
+            XCTFail("Failed to retrieve zip64 format entry from archive"); return
+        }
+        let entry3OriginalOffset = archive["data3.random"]?.zip64ExtendedInformation?.relativeOffsetOfLocalHeader ?? 0
+        do {
+            try archive.remove(entry2)
+        } catch {
+            XCTFail("Failed to remove entry from archive with error : \(error)")
+        }
+        XCTAssert(archive.checkIntegrity())
+        // Should delete zip64 extended information of data3.random as offset changed
```

Comment:
nit: `zip64` should be replaced with `ZIP64` in those comments. (There are multiple occurrences of that string)